PM: Fix device resume order

ceolin opened the pull request:
It fixes a problem that was re-introduced in 81f1225040d317cc9c01ab7d5ef81e098b9ab6e8. 

Added a test to avoid it to happen again.

Changes done after initial review:

 - Tested moved from `device_wakeup_api` to `power_mgmt`
 - Change `power_mgmt` to run only in native posix since it is implementing everything in the application without using anything from platform and because there already is the test `power_mgmt_soc` for properly testing platform integration

Commit:
pm: Fix device resume order

This fix was previously done in 82a6104808bcaeea164e1f41b01d0c07fd9ec3a6
but the commit 81f1225040d317cc9c01ab7d5ef81e098b9ab6e8 re-introduced
the problem.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

gmarull commented on tests/subsys/pm/device_wakeup_api/src/main.c:
```diff
@@ -13,9 +13,106 @@
+		NULL);
+
+
+static int device_b_pm_control(const struct device *dev,
```
ditto

gmarull commented on tests/subsys/pm/device_wakeup_api/src/main.c:
```diff
@@ -13,9 +13,106 @@
 static uint8_t sleep_count;
 
 
+/*
```
note: this test is not really about the wakeup API, it's related to the system PM, better not mix test categories

ceolin replied:
I plan to rename this test to be honest. I think we should avoid having multiple small applications to test only thing we can one application to test multiple things.

ceolin replied:
anyway, will split this one, since both depends on the system PM is hard to de-couple them  keeping it simple to understand.

Commit:
tests: pm: Change power_mgmt test target

This test implements power management entirely in the application, it is
not using anything from the platform so it can be tested using only
native_posix. Also, there already exists power_mgmt_soc to test power
management on real HW.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

Commit:
tests: pm: power_mgmt: Change global variable name

Using a more descriptive name for a global variable since "dev" is
widely used in several APIs signature.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

Commit:
tests: pm: Test devices suspend/resume order

Check if devices are suspend and resumed following the device
initialization order.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

gmarull commented on tests/subsys/pm/power_mgmt/src/main.c:
```diff
@@ -23,12 +23,122 @@ static bool notify_app_exit;
+		PRE_KERNEL_2, CONFIG_KERNEL_INIT_PRIORITY_DEVICE,
+		NULL);
+
+static int device_c_pm_control(const struct device *dev,
```
nit: `pm_action`

gmarull commented on tests/subsys/pm/power_mgmt/boards/native_posix.overlay:
```diff
@@ -0,0 +1,22 @@
+ */
+
+/ {
+	deviceA: deviceA {
```
nit: `device_a`

gmarull commented on tests/subsys/pm/power_mgmt/src/main.c:
```diff
@@ -23,12 +23,122 @@ static bool notify_app_exit;
+		enum pm_device_action pm_action)
+{
+
+	enum pm_device_state stateA;
```
`state_a`

gmarull commented on tests/subsys/pm/power_mgmt/boards/native_posix.overlay:
```diff
@@ -0,0 +1,22 @@
+/ {
+	deviceA: deviceA {
+		label = "deviceA";
+		compatible = "device-a";
```
does `device-a` compatible exist? better create a dummy compatible in `test` bindings folder, e.g. `vnd,pm.yaml`.

ceolin replied:
I'll be consistent with other tests and use "test-device-pm". Following:

`test-gpio-basic-api`
`test-regulator-fixed`
`test-pwm-loopback`

gmarull commented on tests/subsys/pm/power_mgmt/src/main.c:
```diff
@@ -23,12 +23,121 @@ static bool notify_app_exit;
+	return 0;
+}
+
+static int device_a_pm_control(const struct device *dev,
```
this and others: rename to `pm_action` (`pm_control` is no longer used)
